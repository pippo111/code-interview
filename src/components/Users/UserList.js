import React from 'react'
import PropTypes from 'prop-types'

import UserItem from './UserItem'
import { StyledList } from '../Styled'

const UserList = ({ users = [] }) => (
  <StyledList>
    <ol>
      {users.map(user => <li key={user.id}><UserItem {...user} /></li>)}
    </ol>
  </StyledList>
)

UserList.propTypes = {
  users: PropTypes.array
}

export default UserList
