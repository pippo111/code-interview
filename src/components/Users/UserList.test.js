/* global describe, test, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import UserList from './UserList'

const users = [
  {
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz'
  },
  {
    id: 2,
    name: 'Ervin Howell',
    username: 'Antonette',
    email: 'Shanna@melissa.tv'
  }
]

describe('UserList', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<UserList users={users} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders list of users', () => {
    const component = mount(<UserList users={users} />)

    expect(component.find('li').length).toEqual(users.length)
  })
})
