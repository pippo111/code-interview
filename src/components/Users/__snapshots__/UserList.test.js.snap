// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList snapshot renders 1`] = `
<section
  className="sc-AykKC hldZen"
>
  <ol>
    <li>
      <span
        className="main-content"
      >
        Leanne Graham
      </span>
      <span
        className="sub-content"
      >
        @
        Bret
      </span>
    </li>
    <li>
      <span
        className="main-content"
      >
        Ervin Howell
      </span>
      <span
        className="sub-content"
      >
        @
        Antonette
      </span>
    </li>
  </ol>
</section>
`;
